src/libmongoc/examples/example-manage-search-indexes.c

```diff
@@ -0,0 +1,201 @@
+               __LINE__,                                        \
+               #stmt);                                          \
+      abort ();                                                 \
+   } else                                                       \
+      (void) 0
+
+int
+main (int argc, char *argv[])
+{
+   mongoc_client_t *client = NULL;
+   const char *uri_string =
+      "mongodb://127.0.0.1/?appname=create-search-indexes-example";
+   mongoc_uri_t *uri = NULL;
+   mongoc_collection_t *coll = NULL;
+   bson_error_t error;
+   bool ok = false;
+
+   mongoc_init ();
+
+   if (argc > 2) {
+      HANDLE_ERROR (
+         "Unexpected arguments. Expected usage: %s [CONNECTION_STRING]",
+         argv[0]);
+   }
+
+   if (argc > 1) {
+      uri_string = argv[1];
+   }
+
+   uri = mongoc_uri_new_with_error (uri_string, &error);
+   if (!uri) {
+      HANDLE_ERROR ("Failed to parse URI: %s", error.message);
+   }
+   client = mongoc_client_new_from_uri_with_error (uri, &error);
+   if (!client) {
+      HANDLE_ERROR ("Failed to create client: %s", error.message);
+   }
+
+   // Create a random collection name.
+   char collname[25];
+   {
+      // There is a server-side limitation that prevents multiple search indexes
+      // from being created with the same name, definition and collection name.
+      // Atlas search index management operations are asynchronous. Dropping a
+      // collection may not result in the index being dropped immediately. Use a
+      // randomly generated collection name to avoid errors.
+      bson_oid_t oid;
+      bson_oid_init (&oid, NULL);
+      bson_oid_to_string (&oid, collname);
+   }
+
+   // Create collection object.
+   {
+      // Create the collection server-side to avoid the server error:
+      // "Collection 'test.<collname>' does not exist."
+      mongoc_database_t *db = mongoc_client_get_database (client, "test");
+      coll = mongoc_database_create_collection (
+         db, collname, NULL /* options */, &error);
+      if (!coll) {
+         mongoc_database_destroy (db);
+         HANDLE_ERROR ("Failed to create collection: %s", error.message);
+      }
+      mongoc_database_destroy (db);
+   }
+
+   {
+      // Create an Atlas Search Index ... begin
+      bson_t cmd;
+      // Create command.
+      {
+         char *cmd_str = bson_strdup_printf (
+            BSON_STR ({
+               "createSearchIndexes" : "%s",
+               "indexes" : [ {
+                  "definition" : {"mappings" : {"dynamic" : false}},
+                  "name" : "test-index"
+               } ]
+            }),
+            collname);
+         ASSERT (bson_init_from_json (&cmd, cmd_str, -1, &error));
+         bson_free (cmd_str);
+      }
+      if (!mongoc_collection_command_simple (
+             coll, &cmd, NULL /* read_prefs */, NULL /* reply */, &error)) {
+         bson_destroy (&cmd);
+         HANDLE_ERROR ("Failed to run createSearchIndexes: %s", error.message);
```

Comment:
If this example is run against a non-Atlas server, this example fails on this command attempt and error with a "no such cmd" message. Can we consider adding an earlier "example must be run against Atlas" check and error to make the required conditions a bit more clear?

Reply:
I like that idea. ee7806230ad4b84a73e5b75ff256785fe02d262c adds an earlier check that `$listSearchIndexes` is supported, and prints an error suggesting the example may not be using Atlas if an error occurred.